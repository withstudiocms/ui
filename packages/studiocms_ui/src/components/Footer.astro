---
import type { HTMLAttributes } from 'astro/types';

interface FooterLink {
  /**
   * The label of the link.
   */
  label: string;
  /**
   * The href of the link.
   */
  href: string;
}

/**
 * The props for the footer component.
 */
interface Props extends HTMLAttributes<'footer'> {
  /**
   * The links to display in the footer.
   */
	links: {
		[label: string]: FooterLink[];
	};
  /**
   * The copyright text to display in the footer.
   */
	copyright: string;
}

const { copyright, links, ...props } = Astro.props;
---

<footer {...props}>
  <div class="upper">
    <div>
      <slot name="brand" />
    </div>
    <div class="links">
      {Object.keys(links).map((groupLabel) => (
        <ul>
          <li class="sui-footer-link-label">{groupLabel}</li>
          {links[groupLabel]!.map((item) => (
            <li>
              <a href={item.href}>{item.label}</a>
            </li>
          ))}
        </ul>
      ))}
    </div>
  </div>
  <hr class="separator" />
  <div class="lower">
    <span class="copyright-span">{copyright}</span>
    <slot name="socials" />
  </div>
</footer>
<style>
  footer {
    display: flex;
    flex-direction: column;
    gap: 2rem;
    background-color: hsl(var(--background-step-1));
    padding: 2rem 10vw;
    color: hsl(var(--text-normal)) !important;
  }

  .upper, .lower {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
  }

  .links {
    display: flex;
    justify-content: flex-end;
    flex-direction: row;
    flex-wrap: wrap;
    gap: 2rem;
  }

  .links ul {
    list-style-type: none;
    margin: 0 !important;
    padding: 0 !important;
  }

  .links ul li, .links ul li * {
    color: hsl(var(--text-normal)) !important;
    width: fit-content;
  }

  .links ul li:has(a):hover {
    text-decoration: underline;
  }

  .sui-footer-link-label {
    font-size: 1.125em;
    font-weight: 700;
  }

  .separator {
    height: 1px;
    width: 100%;
    border: none;
    background: hsl(var(--border));
  }

  .lower {
    align-items: center;
    flex-wrap: wrap;
    gap: 1rem;
  }

  @media screen and (max-width: 1440px) {
    footer {
      padding: 2rem;
    }
  }

  @media screen and (max-width: 1280px) {
    .upper {
      flex-direction: column;
      gap: 2rem;
    }

    .links {
      justify-content: flex-start;
    }
  }

  @media screen and (max-width: 640px) {
    .links ul {
      width: calc(50% - 1rem);
    }
  }
</style>
