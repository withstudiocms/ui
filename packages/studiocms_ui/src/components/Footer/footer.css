footer.sui-footer {
  display: flex;
  flex-direction: column;
  gap: 2rem;
  background-color: hsl(var(--background-step-1));
  padding: 2rem 10vw;
  color: hsl(var(--text-normal)) !important;
}

.upper, .lower {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.links {
  display: flex;
  justify-content: flex-end;
  flex-direction: row;
  flex-wrap: wrap;
  gap: 2rem;
}

.links ul {
  list-style-type: none;
  margin: 0 !important;
  padding: 0 !important;
}

.links ul li, .links ul li * {
  color: hsl(var(--text-normal)) !important;
  width: fit-content;
}

.links ul li:has(a):hover {
  text-decoration: underline;
}

.sui-footer-link-label {
  font-size: 1.125em;
  font-weight: 700;
}

.separator {
  height: 1px;
  width: 100%;
  border: none;
  background: hsl(var(--border));
}

.lower {
  align-items: center;
  flex-wrap: wrap;
  gap: 1rem;
}

@media screen and (max-width: 1440px) {
  footer.sui-footer {
    padding: 2rem;
  }
}

@media screen and (max-width: 1280px) {
  .upper {
    flex-direction: column;
    gap: 2rem;
  }

  .links {
    justify-content: flex-start;
  }
}

@media screen and (max-width: 640px) {
  .links ul {
    width: calc(50% - 1rem);
  }
}
