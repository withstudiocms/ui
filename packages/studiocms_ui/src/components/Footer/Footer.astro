---
import type { HTMLAttributes } from 'astro/types';
import './footer.css';

interface FooterLink {
	/**
	 * The label of the link.
	 */
	label: string;
	/**
	 * The href of the link.
	 */
	href: string;
}

/**
 * The props for the footer component.
 */
interface Props extends HTMLAttributes<'footer'> {
	/**
	 * The links to display in the footer.
	 */
	links: {
		[label: string]: FooterLink[];
	};
	/**
	 * The copyright text to display in the footer.
	 */
	copyright: string;
}

const { copyright, class: className, links, ...props } = Astro.props;
---

<footer {...props} class={`${className} sui-footer`}>
  <div class="upper">
    <div>
      <slot name="brand" />
    </div>
    <div class="links">
      {Object.keys(links).map((groupLabel) => (
        <ul>
          <li class="sui-footer-link-label">{groupLabel}</li>
          {links[groupLabel]!.map((item) => (
            <li>
              <a href={item.href}>{item.label}</a>
            </li>
          ))}
        </ul>
      ))}
    </div>
  </div>
  <hr class="separator" />
  <div class="lower">
    <span class="copyright-span">{copyright}</span>
    <slot name="socials" />
  </div>
</footer>
