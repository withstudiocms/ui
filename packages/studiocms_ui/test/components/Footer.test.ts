import { describe, expect } from 'vitest';
import Footer from '../../src/components/Footer/Footer.astro';
import { test } from '../fixtures/vitest/AstroContainer';

const mockProps = {
	links: {
		Resources: [
			{ label: 'Documentation', href: '#' },
			{ label: 'Tutorials', href: '#' },
		],
		Legal: [
			{ label: 'Privacy Policy', href: '#' },
			{ label: 'Terms of Service', href: '#' },
		],
	},
	copyright: 'Â© 2024 Your Company. All rights reserved.',
};

const slots = {
	brand: 'YourCompany',
};

describe('Footer Component', () => {
	test('renders Footer component correctly', async ({ renderComponent }) => {
		const result = await renderComponent(Footer, 'Footer', { props: mockProps, slots });
		expect(result).toMatchSnapshot();
	});
});
