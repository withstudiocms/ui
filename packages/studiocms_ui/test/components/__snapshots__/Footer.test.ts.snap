// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer Component > renders Footer component correctly 1`] = `"<footer class="undefined sui-footer"> <div class="upper"> <div> YourCompany </div> <div class="links"> <ul> <li class="sui-footer-link-label">Resources</li> <li> <a href="#">Documentation</a> </li><li> <a href="#">Tutorials</a> </li> </ul><ul> <li class="sui-footer-link-label">Legal</li> <li> <a href="#">Privacy Policy</a> </li><li> <a href="#">Terms of Service</a> </li> </ul> </div> </div> <hr class="separator"> <div class="lower"> <span class="copyright-span">Â© 2024 Your Company. All rights reserved.</span>  </div> </footer>"`;
